src/network/deployer.ts

```diff
@@ -0,0 +1,270 @@
+import '@nomiclabs/hardhat-ethers';
+import '@openzeppelin/hardhat-upgrades';
+import path from 'path';
+import hre from 'hardhat';
+import ethereum from './providers/ethereum';
+import {getFile, uploadFile} from '../client/storage';
+import logger from '../core/log';
+import BN from 'bn.js';
+import {Artifact} from 'hardhat/types';
+import os from 'os';
+import fs from 'fs-extra';
+import axios from 'axios';
+import tar from 'tar-fs';
+import gunzip from 'gunzip-maybe';
+const log = logger.child({module: 'Deployer'});
+
+const IS_PACKAGED = Boolean((process as typeof process & {pkg?: unknown}).pkg);
+const PROXY_METADATA_KEY = 'zweb/contracts/proxy/metadata';
+
+// TODO: this is a temporary solution, supporting only limited number of dependencies
+const NPM_DEPS_URLS: Record<string, string> = {
+    '@openzeppelin/contracts': 'https://registry.npmjs.org/@openzeppelin/contracts/-/contracts-4.7.3.tgz',
```

Comment:
I think the dependencies should also be read from blockchain or storage (maybe we should upload a sort of dependency list or a `package.json` file). Otherwise we'll have to release new Engine every time these dependencies change.

Reply:
Again, this is the temporary solution, ideally we should add a function, which resolves this link using npm package name, but the goal was to make it work ASAP